core/rewards/staking_reward_calculator_test.go

```diff
@@ -63,6 +63,19 @@ func testCalcRewardsZeroScores(t *testing.T) {
 	require.Equal(t, 0, len(res.partyToAmount))
 }
 
+func TestFilterZeros(t *testing.T) {
+	delegatorShare, _ := num.DecimalFromString("0.3")
+	scores := map[string]num.Decimal{}
+	scores["node1"] = num.NewDecimalFromFloat(1)
+	scores["node2"] = num.DecimalZero()
+	scores["node3"] = num.DecimalZero()
+	scores["node4"] = num.DecimalZero()
+
+	res := calculateRewardsByStake("1", "asset", "rewardsAccountID", num.NewUint(100000), scores, []*types.ValidatorData{{NodeID: "node1", StakeByDelegators: num.NewUint(500), SelfStake: num.NewUint(1000), Delegators: map[string]*num.Uint{"zohar": num.UintZero(), "jeremy": num.NewUint(500)}}}, delegatorShare, num.UintZero(), rng, logging.NewTestLogger())
+	require.Equal(t, num.NewUint(100000), res.totalReward)
+	require.Equal(t, 2, len(res.partyToAmount))
```

Comment:
Can we add a party with 0 delegation there and  verify that there are `zohar` and `jeremy` specifically ?

With this verification, I don't think we can't ensure it's correct.

Reply:
I disagree. without this fix the count would return 3, now it returns 2. I am already a party with 0 delegation.

Reply:
but sure, can add verification by party to demonstrate that "zohar" is not there.